test/services/embeds.test.js

```diff
@@ -1,10 +1,55 @@
 const assert = require('assert');
 const app = require('../../src/app');
+const service = app.service('embeds');
+
+const port = app.get('port') || 3030;
 
 describe('\'embeds\' service', () => {
-  it('registered the service', () => {
-    const service = app.service('embeds');
+  before(function(done) {
+    this.server = app.listen(port);
+    this.server.once('listening', () => done());
+  });
 
+  after(function(done) {
+    this.server.close(done);
+  });
+
+  it('registered the service', () => {
     assert.ok(service, 'Registered the service');
   });
+
+  // Parameters for the find() method on the Service class.
+  let params = {
+    simpleUrl: {
+      query: {
+        url: 'https://www.youtube.com/watch?v=QWU9YsxXPqw'
+      }
+    },
+    startTimeUrl: {
+      query: {
+        url: 'https://www.youtube.com/watch?v=QWU9YsxXPqw&start=332'
+      }
+    }
+  };
+
+  it('GET /embeds?url={simpleUrl}', async () => {
+    // Calls Embeds Service.find(params) method, which will trigger the URL validation
+    await service.find(params.simpleUrl).then((body) => {
```

Comment:
And part 2/2 of what I would like to see changed before merge: Use either `await/async` *or* `then`. I would prefer `await/async`.
```js
try {
    let body = await service.find(params.simpleUrl)
    assert.strictEqual(body.site_name.toLowerCase(), 'youtube');
    ...
} catch (err) {
    return err
}
```

And just use `strictEqual` or `deepEqual` whenever you want to use `equal`. I think there was even some deprecation around.

See https://nelsonic.gitbooks.io/node-js-by-example/content/core/assert/README.html for the difference.

Reply:
https://github.com/Human-Connection/Embed-API/pull/19/commits/745d855fc4e42be5e6bb5bd98c46416bc0e638fb